nixpkgs_review/buildenv.py

```diff
@@ -20,11 +19,32 @@ def find_nixpkgs_root() -> Optional[str]:
         prefix.append("..")
 
 
-@dataclass
 class Buildenv:
-    allow_aliases: bool
+    def __init__(self, allow_aliases: bool, extra_nixpkgs_config: str) -> None:
+        if not (
+            extra_nixpkgs_config.startswith("{") and extra_nixpkgs_config.endswith("}")
+        ):
+            raise RuntimeError(
+                "--extra-nixpkgs-config should start with `{` and end with `}`"
+            )
+
+        self.nixpkgs_config = NamedTemporaryFile(suffix=".nix")
+        self.nixpkgs_config.write(
+            str.encode(
+                f"""{{
+  allowUnfree = true;
+  allowBroken = true;
+  {"allowAliases = false;" if not allow_aliases else ""}
+  checkMeta = true;
+  ## TODO: also build packages marked as insecure
+  # allowInsecurePredicate = x: true;
+}} // {extra_nixpkgs_config}
+"""
+            )
+        )
+        self.nixpkgs_config.flush()
 
-    def __enter__(self) -> None:
+    def __enter__(self) -> str:
```

Comment:
Maybe make it `Path`? It's understandable here, but gets confusing by the time it reaches `review_local_revision(..., nixpkgs_config: str, ...)`